frontend/document_collaboration_system/src/index.js

```diff
@@ -0,0 +1,128 @@
+import React, { createContext } from "react";
+import ReactDOM from "react-dom";
+import "./css/index.css";
+import App from "./App";
+import Home from "./home";
+import DocumentList from "./documentList";
+import Login from "./login";
+import reportWebVitals from "./reportWebVitals";
+import "react-draft-wysiwyg/dist/react-draft-wysiwyg.css";
+import {
+  BrowserRouter as Router,
+  Switch,
+  Route,
+  Redirect,
+} from "react-router-dom";
+import CustomizedSnackbars from "./customAlert";
+import InviteSnackbar from "./inviteAlert";
+import { api } from "./service";
+import Messages from "./chat";
+import { Error404 } from "./error";
+import Header from "./header";
+import Navbar from "./navbar";
+import Footer from "./footer";
+
+export const AppContext = createContext();
+
+class Index extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      nickname: "",
+      session_token: "",
+      document: "",
+      customAlert: { alertOpen: false, alertMessage: "", type: "" },
+      notification: {
+        open: false,
+        setOpen: null,
+        documentName: "",
+        inviteId: "",
+      },
+    };
+
+    this.updateState = (field) => (value) => {
+      this.setState({ [field]: value });
+    };
+
+    this.loadNotification = () => {
+      api.getInvite().then((response_data) => {
+        if (response_data !== null) {
+          this.setState({
+            notification: {
+              open: true,
+              setOpen: this.updateState("notification"),
+              documentName: response_data.document_name,
+              documentId: response_data.document_id,
+              inviteId: response_data.invite_id,
+            },
+          });
+        }
+      });
+    };
+  }
+
+  componentDidMount() {
+    this.updateState("session_token")(sessionStorage.getItem("token"));
```

Comment:
you dont need updateState function
just this.setState{{
session_token: sessionStorage.getItem("token"),
username: sessionStorage.getItem("username")
}}
one update instead of 2